pkgs/state-transition-runtime/build.zig

```diff
@@ -39,26 +57,42 @@ pub fn build(b: *Builder) void {
     // target has to be riscv5 runtime provable/verifiable on zkVMs
     const exe = b.addExecutable(.{
         .name = "zeam-state-transition-runtime",
-        .root_source_file = .{ .cwd_relative = "src/main.zig" },
+        .root_source_file = b.path("src/main.zig"),
         .optimize = optimize,
         .target = target,
     });
     // addimport to root module is even required afer declaring it in mod
     exe.root_module.addImport("ssz", ssz);
     exe.root_module.addImport("zeam-types", zeam_types);
     exe.root_module.addImport("zeam-state-transition", zeam_state_transition);
+    exe.root_module.addImport("zkvm", zkvm_module);
+    switch (zkvm) {
+        .ceno => {
+            exe.addAssemblyFile(b.path("src/ceno/start.s"));
+            exe.setLinkerScript(b.path("src/ceno/ceno.ld"));
+        },
+        .powdr => {
+            exe.addAssemblyFile(b.path("src/powdr/start.s"));
+            exe.setLinkerScript(b.path("src/powdr/powdr.x"));
+            exe.pie = true;
+        },
+        .sp1 => {
+            exe.addAssemblyFile(b.path("src/sp1/start.s"));
+            exe.setLinkerScript(b.path("src/sp1/sp1.ld"));
+        },
+    }
     b.installArtifact(exe);
 
-    const tests = b.addTest(.{
-        .root_source_file = .{ .cwd_relative = "src/main.zig" },
-        .optimize = optimize,
-        .target = target,
-    });
-    tests.root_module.addImport("ssz", ssz);
-    tests.root_module.addImport("zeam-types", zeam_types);
-    tests.root_module.addImport("zeam-transition-runtime", zeam_state_transition);
+    // const tests = b.addTest(.{
+    //     .root_source_file = .{ .cwd_relative = "src/transition.zig" },
+    //     .optimize = optimize,
+    //     .target = target,
+    // });
+    // tests.root_module.addImport("ssz", ssz);
+    // tests.root_module.addImport("zeam-types", zeam_types);
+    // tests.root_module.addImport("zeam-transition-runtime", zeam_state_transition);
 
-    const run_tests = b.addRunArtifact(tests);
-    const test_step = b.step("test", "Run unit tests");
-    test_step.dependOn(&run_tests.step);
+    // const run_tests = b.addRunArtifact(tests);
+    // const test_step = b.step("test", "Run unit tests");
+    // test_step.dependOn(&run_tests.step);
```

Comment:
I'm commenting this out, because these tests can never run (unless we are building this on riscv32). It might be a good idea to have the CI build them and run them in qemu, but this is an effort that is quite involved, and it can be tested another way by building the executable and running a prover instead. So I might end up removing it all, I just need to think about the best way to do this.